### DISCUSSION TITLE
Calibrations not working correctly

### DISCUSSION THREAD

**Discussion by GLoren1** [NightscoutFoundation/xDrip#3550] [Q&A]
I have often had problems with the way xDrip deals with calibrations.  For example, I restarted a G6 sensor yesterday, and after warming up, the readings were about 80% higher than actual, as expected.  After waiting about 24 hours, I started to add calibrations at 20 % intervals.  First I added a calibration that was about 20% less than the reading, and after about 30 minutes, xDrip reverted back to the value it had before calibration (20% higher).  I waited another hour, and tried the same thing--added a calibration 20% less than the reading.  Again xDrip reverted back after about 25 mintues.  I repeated this 6-7 times over the course of a couple of hours and was never able to get the calibration to work. 

 Eventually, I had to insert a new sensor because I couldn't wait any more with incorrect readings.

**Comment by Navid200** [comment]
Can you provide screenshots of the main screen when you did the calibration as well as a screenshot of the Dex status page?

Alternatively, I can make a statement here regardless.  xDrip does not calibrate G6.  xDrip submits calibrations for G6 (or G7) to the transmitter.  This is because xDrip runs G6 or G7 in native mode.  
I have gone over this here: https://navid200.github.io/xDrip/docs/Native-Algorithm.html

It is possible you may be running your G6 in non-native mode.  That is why I asked to see the status page to verify that.  If that is the case, we can talk about why xDrip misbehaves.  
Otherwise, xDrip has nothing to do with reverting back after a calibration.

**Reply by GLoren1** [reply]
Thanks. I've I've attached the screenshots 

![Screenshot_20240629-164532_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/62320462/59f59660-e417-45a4-ad9b-0489cadb268f)
![Screenshot_20240629-164838_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/62320462/810555c7-73e9-490c-a6f0-608a1e559c16)

**Reply by Navid200** [reply]
That transmitter can only run in Native mode.  In that case, calibrations are submitted to the transmitter by xDrip.   
We don't recommend restarting a G6.

But, if you decide to do it, these are the instructions: https://navid200.github.io/xDrip/docs/Restart-G6-sensor.html

**Reply by Navid200** [reply]
And you are running an xDrip release older than our current stable release.  You should update.

**Reply by GLoren1** [reply]
> That transmitter can only run in Native mode. In that case, calibrations are submitted to the transmitter by xDrip.

Thanks. Can you clarify what you mean by this--specifically, does this effect my calibration issue, and are you recommending that I adjust anything?

**Reply by Navid200** [reply]
No, sorry  

I just wanted to make a complete statement.  Saying that you can run G6 only in native mode is not a 100% valid statement.
If you had a very old G6 or a modified G6, you could run them in non-native mode, which would not be recommended anyway. 
In that case, xDrip would be responsible for calibrations.  
You should run a G6 in native mode even if you had a G6 you could run in non-native mode.
The reason is that G6 offers factory calibration, which is valid only in native mode.  

If you use a G6 with a new sensor and follow the instructions meaning you start the sensor as soon as you insert it and use the correct 4-digit calibration code, the G6 error will be minimal and comparable to the error of your blood glucose meter.  In that case, even if there is a delta between G6 and your blood glucose meter, you should not calibrate.
You should only calibrate a G6 if you are certain something is wrong with it for example if you have symptoms that suggest it is wrong.

The reason we don't recommend restarting a G6 is that factory calibration becomes invalid.
That's why you will then need to calibrate after restart.
But then, you have to deal with what the transmitter does with your submitted calibrations.
